AsyncDisplayKit/Details/Transactions/_ASAsyncTransaction.mm

```diff
@@ -159,32 +207,29 @@ - (NSString *)description
   Operation operation;
   operation._block = block;
   operation._group = this;
-  entry._operations.push_back(operation);
+  operation._priority = priority;
+  entry.pushOperation(operation);
   
   ++_pendingOperations; // enter group
   
-  NSUInteger maxThreads = [NSProcessInfo processInfo].activeProcessorCount;
-  if (maxThreads < 2) { // it is reasonable to have at least two working threads, also
-    maxThreads = 2;     // [_ASDisplayLayerTests testTransaction] requires at least two threads
-  }
+  NSUInteger maxThreads = [NSProcessInfo processInfo].activeProcessorCount * 2;
 
-#if 0
-  // Bit questionable - we can give main thread more CPU time during tracking;
-  if (maxThreads > 1 && [[NSRunLoop mainRunLoop].currentMode isEqualToString:UITrackingRunLoopMode])
+  // Bit questionable maybe - we can give main thread more CPU time during tracking;
```

Comment:
Yeah. This way you get 3 threads on double core CPU while tracking, 4 threads while still. It can always be tweaked, but this seems like a decent compromise in my case. Still, if you play with it and find something that works better for pinterest I'm all ears